### PR TITLE
[FLINK-17850][jdbc] Fix PostgresCatalogITCase.testGroupByInsert() fails on CI

### PR DESCRIPTION
**Pull request by wuchong**
## What is the purpose of the change

Sorry for breaking the CI builds. The reason is that #12090 updates the schema of the base testing table. But I didn't rerun the builds before merging #11906 which uses this testing table. 
This PR fix the new added test in #11906.

## Brief change log

- Update the `testGroupByInsert()` to add the new column `decimal`.

## Verifying this change

This change is already covered by existing tests.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by wuchong** [comment]
cc @leonardBang @fpompermaier , could you help to have a look at this?

File: flink-connectors/flink-connector-jdbc/src/test/java/org/apache/flink/connector/jdbc/catalog/PostgresCatalogITCase.java [hunk 1]
```diff
@@ -69,18 +69,39 @@ public void test_fullPath() throws Exception {
...
 
 		List<Row> results = Lists.newArrayList(
-			tEnv.sqlQuery(String.format("select * from %s", TABLE1)).execute().collect());
+			tEnv.sqlQuery(String.format("select * from %s", TABLE4)).execute().collect());
 		assertEquals("[1]", results.toString());
 	}
 
+	@Test
+	public void testGroupByInsert() {
+		TableEnvironment tEnv = getTableEnvWithPgCatalog();
```
Comment (leonardBang) [inline flink-connectors/flink-connector-jdbc/src/test/java/org/apache/flink/connector/jdbc/catalog/PostgresCatalogITCase.java:86]: How about add a before method to reuse tEnv cause every test use it ?

File: flink-connectors/flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/catalog/AbstractJdbcCatalog.java [hunk 2]
```diff
@@ -126,31 +124,29 @@ public String getBaseUrl() {
...
-		if (columnsWithIndex != null) {
-			// sort columns by KEY_SEQ
-			columnsWithIndex.sort(Comparator.comparingInt(Map.Entry::getKey));
-			List<String> cols = columnsWithIndex.stream().map(Map.Entry::getValue).collect(Collectors.toList());
-			return UniqueConstraint.primaryKey(pkName, cols);
+		List<String> pkFields = Arrays.asList(new String[keySeqColumnName.size()]); // initialize size
+		keySeqColumnName.forEach(pkFields::set);
+		if (!pkFields.isEmpty()) {
+			// PK_NAME maybe null according to the javadoc, generate an unique name in that case
+			pkName = pkName != null ? pkName : "pk_" + String.join("_", pkFields);
```
Comment (leonardBang) [inline flink-connectors/flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/catalog/AbstractJdbcCatalog.java:146]:
```suggestion
			pkName = pkName == null ? "pk_" + String.join("_", pkFields) : pkName;
```
How about this?

**Comment by fpompermaier** [comment]
LGTM on my side. Then what should I do with the other PR that was reverted?

**Comment by wuchong** [comment]
Thanks for the reviewing @fpompermaier .  This is the only commit been reverted.  I will merge this when the build is passed.

**Comment by leonardBang** [comment]
Thanks for the update @wuchong , LGTM

**Pull request closed without merging**